src/Service/Assembler/SponsorshipBenefitAssembler.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Service\Assembler;
+
+use App\Entity\SponsorshipBenefit;
+use App\Service\DTO\SponsorshipBenefitData;
+use Ramsey\Uuid\Uuid;
+
+final class SponsorshipBenefitAssembler implements AssemblerInterface
+{
+    /**
+     * @param SponsorshipBenefitData $entityData
+     * @param ?SponsorshipBenefit $entity
+     * @return SponsorshipBenefit
+     * @throws \Exception
+     */
+    public function read($entityData, $entity = null)
+    {
+        if (!$entity) {
+            $entity = new SponsorshipBenefit();
+            $entity->setId(Uuid::uuid4());
```

Comment:
[Objects should be created in one go](https://matthiasnoback.nl/2018/07/objects-should-be-constructed-in-one-go/)
You're opening your entity to an invalid state (the entity does not have an ID until someone calls that method).